server/src/test/java/org/elasticsearch/common/UUIDTests.java

```diff
@@ -237,4 +240,13 @@ public void testStringLength() {
     private static int getUnpaddedBase64StringLength(int sizeInBytes) {
         return (int) Math.ceil(sizeInBytes * 4.0 / 3.0);
     }
+
+    private void verifyUUIDIsUrlSafe(final String uuid) {
+        assertFalse("UUID should not contain padding characters: " + uuid, uuid.contains("="));
+        try {
+            BASE_64_URL_DECODER.decode(uuid);
```

Comment:
I may've missed it, but it'd be nice to test that encoding and decoding returns the original byte array, for random inputs.

Reply:
+1 to such a test